File: archive/canvas-api/src/main/java/com/instructure/canvasapi/api/AvatarAPI.java
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */

package com.instructure.canvasapi.api;

import com.instructure.canvasapi.model.Avatar;
import com.instructure.canvasapi.model.User;
import com.instructure.canvasapi.utilities.APIHelpers;
import com.instructure.canvasapi.utilities.CanvasCallback;
import com.instructure.canvasapi.utilities.CanvasRestAdapter;

import retrofit.Callback;
import retrofit.RestAdapter;
import retrofit.http.Body;
import retrofit.http.GET;
import retrofit.http.PUT;
import retrofit.http.Query;


public class AvatarAPI extends BuildInterfaceAPI {

    interface AvatarsInterface{
        @GET("/users/self/avatars")
        void getFirstPageOfAvatarList( Callback<Avatar[]> callback);

        @PUT("/users/self")
        void updateAvatar(@Query("user[avatar][url]") String avatarURL, @Body String body, Callback<User> callback);
    }

    /////////////////////////////////////////////////////////////////////////
    // API Calls
    /////////////////////////////////////////////////////////////////////////

    public static void getFirstPageOfAvatarList(CanvasCallback<Avatar[]> callback){
        if(APIHelpers.paramIsNull(callback)) { return; }

        buildInterface(AvatarsInterface.class, callback).getFirstPageOfAvatarList( callback);
        buildInterface(AvatarsInterface.class, callback).getFirstPageOfAvatarList( callback);
    }

    public static void updateAvatar(String avatarURL, CanvasCallback<User> callback){
        if(APIHelpers.paramIsNull(callback,avatarURL)){ return; }

        buildInterface(AvatarsInterface.class, callback).updateAvatar(avatarURL, "", callback);
    }

}

File: libs/canvas-api-2/src/main/java/com/instructure/canvasapi2/apis/AvatarAPI.kt
/*
 * Copyright (C) 2017 - present Instructure, Inc.
 *
 *     Licensed under the Apache License, Version 2.0 (the "License");
 *     you may not use this file except in compliance with the License.
 *     You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 *     Unless required by applicable law or agreed to in writing, software
 *     distributed under the License is distributed on an "AS IS" BASIS,
 *     WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 *     See the License for the specific language governing permissions and
 *     limitations under the License.
 *
 */
package com.instructure.canvasapi2.apis

import com.instructure.canvasapi2.StatusCallback
import com.instructure.canvasapi2.builders.RestBuilder
import com.instructure.canvasapi2.builders.RestParams
import com.instructure.canvasapi2.models.Avatar
import com.instructure.canvasapi2.models.User

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.PUT
import retrofit2.http.Query

object AvatarAPI {

    internal interface AvatarsInterface {
        @PUT("users/self")
        fun updateAvatar(@Query("user[avatar][url]") avatarUrl: String): Call<User>

        @PUT("users/self")
        fun updateAvatarWithToken(@Query("user[avatar][token]") avatarToken: String): Call<User>
    }

    fun updateAvatar(adapter: RestBuilder, params: RestParams, avatarUrl: String, callback: StatusCallback<User>) {
        callback.addCall(adapter.build(AvatarsInterface::class.java, params).updateAvatar(avatarUrl)).enqueue(callback)
    }

    fun updateAvatarWithToken(adapter: RestBuilder, params: RestParams, avatarToken: String, callback: StatusCallback<User>) {
        callback.addCall(adapter.build(AvatarsInterface::class.java, params).updateAvatarWithToken(avatarToken)).enqueue(callback)
    }
}
